enhancements/per-node-configuration.md

```diff
@@ -0,0 +1,656 @@
+# Per-Node Configuration with Node Selectors
+
+## Summary
+
+This enhancement proposes adding node selector support to all OpenPERouter CRDs (`Underlay`, `L3VNI`, `L2VNI`, and `L3Passthrough`), enabling different nodes to have different network configurations. This allows OpenPERouter to support heterogeneous cluster topologies including multi-datacenter, multi-rack, and mixed-hardware environments where different nodes may need different BGP configurations, VNI assignments, or passthrough settings.
+
+## Motivation
+
+Currently, OpenPERouter applies all CRD configurations cluster-wide, meaning all nodes in the cluster receive the same configuration. This creates operational limitations in real-world deployments:
+
+**Underlay limitations:**
+- **Multi-rack deployments**: Different nodes are physically connected to different ToR (Top of Rack) switches but cannot be configured to peer with their respective ToRs
+- **Multi-datacenter clusters**: Nodes distributed across availability zones or datacenters need to peer with location-specific BGP routers
+- **Hardware heterogeneity**: Different server models have different NIC naming conventions (e.g., Dell uses `eno1`, HP uses `em1`), preventing a one-size-fits-all configuration
+- **Topology flexibility**: Cannot accommodate nodes in different physical network topologies within the same cluster
+
+**L3VNI/L2VNI limitations:**
+- **Per-rack VNI isolation**: Different racks may need separate VNI configurations for network segmentation
+- **Zone-specific VNIs**: Nodes in different availability zones may require different VNI configurations for compliance or performance reasons
+- **Selective VNI deployment**: Some workloads on specific nodes may need access to certain VNIs while others don't
+- **Gateway placement**: L2 gateway IPs may need to be configured only on specific nodes based on network topology
+
+**L3Passthrough limitations:**
+- **Selective passthrough**: Only specific nodes should participate in direct BGP fabric communication
+- **Security zones**: Different security zones may require different passthrough configurations
+- **Specialized workloads**: Only nodes running certain workloads should have passthrough capabilities
+
+### Goals
+
+- Enable per-node configuration for all OpenPERouter CRDs using Kubernetes node selectors
+- Support heterogeneous network configurations across different hardware platforms, racks, and zones
+- Allow multiple instances of each CRD to coexist, each targeting specific node subsets
+- Maintain backward compatibility with existing cluster-wide configurations
+- Prevent configuration conflicts through validation (no node should be matched by multiple instances of the same CRD type)
```

Comment:
wait, I'm confused. For instance, why can't the user have multiple L2 VNIs in the same node ? 

I think this goal is not right - you probably forgot to adjust the text.

Reply:
Right let me updated this.